### PR TITLE
added code fixes

### PR DESCRIPTION
**Pull request by nuel-ikwuoma**
added some code fixes, to remove compilation errors

### CODE REVIEW

File: lib/Test.sol [hunk 1]
```diff
@@ -4,7 +4,7 @@ pragma solidity ^0.8.13;
 import "forge-std/Test.sol";
 import "./MerkleMultiProof.sol";
 import "./MerkleMountainRange.sol";
-import { TrieDB } from "./mpt/TrieDB.sol";
+// import { TrieDB } from "./mpt/TrieDB.sol";
```
Comment (seunlanlege) [inline lib/Test.sol:7]: You can remove

File: lib/trie/NodeCodec.sol [hunk 2]
```diff
@@ -34,11 +34,11 @@ library NodeCodec {
         return node.isInline;
     }
 
-    function loadValue(NodeHandle memory node, HashDB hashDB) external returns (bytes memory) {
+    function loadValue(NodeHandle memory node, TrieDB trieDB) external returns (bytes memory) {
```
Comment (seunlanlege) [inline lib/trie/NodeCodec.sol:37]: Unused method, let's remove

File: lib/trie/TrieDB.sol [hunk 3]
```diff
@@ -6,21 +6,23 @@ import "./Node.sol";
 
 interface TrieDB {
     function get(bytes32 hash) external returns (bytes memory);
-    function length() external returns (uint256);
+    // function length() external returns (uint256);    // Not implemented in substrateTrieDB
```
Comment (seunlanlege) [inline lib/trie/TrieDB.sol:9]: Remove

File: lib/trie/TrieDB.sol [hunk 4]
```diff
@@ -6,21 +6,23 @@ import "./Node.sol";
...
-        } else if (node.isHash) {
-            return get(node.hash);
-        }
+    // default implementation not allowed
+    // function load(NodeHandle memory node) external returns (bytes memory) {
+    //     if (node.isInline) {
+    //         return node.inLine;
+    //     } else if (node.isHash) {
+    //         return this.get(node.hash);
+    //     }
```
Comment (seunlanlege) [inline lib/trie/TrieDB.sol:24]: Let's make it abstract

File: lib/trie/substrate/SubstrateTrieDB.sol [hunk 5]
```diff
@@ -18,21 +18,23 @@ contract SubstrateTrieDB is TrieDB {
     uint8 public constant BRANCH_WITHOUT_MASK = 0x02 << 6;
     uint8 public constant ALT_HASHING_LEAF_PREFIX_MASK = FIRST_PREFIX | (0x01 << 5);
     uint8 public constant ALT_HASHING_BRANCH_WITH_MASK = FIRST_PREFIX | (0x01 << 4);
+    uint8 public constant NIBBLE_PER_BYTE = 2;
     uint256 public constant NIBBLE_SIZE_BOUND = uint256(type(uint16).max);
     uint256 public constant BITMAP_LENGTH = 2;
     uint256 public constant HASH_lENGTH = 32;
 
-    mapping(bytes32 => NodeKind) internal db;
+    // mapping(bytes32 => NodeKind) internal db;
```
Comment (seunlanlege) [inline lib/trie/substrate/SubstrateTrieDB.sol:26]: Remove

File: lib/trie/substrate/SubstrateTrieDB.sol [hunk 6]
```diff
@@ -99,7 +101,7 @@ contract SubstrateTrieDB is TrieDB {
         }
 
         for (uint256 i = 0; i < 16; i ++) {
-            NodeHandleOption handle;
+            // NodeHandleOption handle;     // shadows existing `handle` variable
```
Comment (seunlanlege) [inline lib/trie/substrate/SubstrateTrieDB.sol:104]: Let's rename the variable, we shouldn't re use the old one

**Review by seunlanlege — CHANGES_REQUESTED** [review]
Just a few minor stuff

File: lib/trie/substrate/SubstrateTrieDB.sol (hunk 6 above)
Comment (nuel-ikwuoma) [reply lib/trie/substrate/SubstrateTrieDB.sol:104]: what name would you suggest? im thinking `childHandle`

Comment (seunlanlege) [reply lib/trie/substrate/SubstrateTrieDB.sol:104]: Yeah that works

**Comment by nuel-ikwuoma** [comment]
the changes have been updated

**Review by seunlanlege — APPROVED** [review]

**Pull request merged**